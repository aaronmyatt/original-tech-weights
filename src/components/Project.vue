<template>
<ul class="project-container list-group">
  <li class="project-name card-title list-group-item d-flex w-100 justify-content-between">
    <h5>{{ projectName }}</h5>
  </li>

  <li class="project-description list-group-item d-flex w-100 justify-content-between">
    {{ projectDescription }}
  </li>

  <li class="project-date list-group-item d-flex w-100 justify-content-between">
    {{ projectDate }}
  </li>

  <li class="project-owner list-group-item d-flex w-100 justify-content-between">
    {{ projectOwner }}
  </li>

  <li class="project-repo-url list-group-item d-flex w-100 justify-content-between">
    {{ projectUrl }}
  </li>

  <li class="project-tech-weights list-group-item">
    <WeightVisualiser :weightList="getWeightList()"/>
  </li>
</ul>
</template>

<script>
import WeightVisualiser from './WeightVisualiser';

export default {
  name: 'Project',
  components: {
    WeightVisualiser,
  },
  props: [
    'projectName',
    'projectDescription',
    'projectDate',
    'projectOwner',
    'projectUrl',
    'projectWeights',
  ],
  methods: {
    getWeightList() {
      return this.projectWeights;
    },
  },
};
</script>

<style scoped>
</style>
