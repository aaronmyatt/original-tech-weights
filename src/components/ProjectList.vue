<template>
<ul class="project-list">
  <li class="project card" v-for="project, index in projectList" :key="index">
    <Project v-bind="project" />
  </li>
</ul>
</template>

<script>
import Project from './Project';

export default {
  name: 'ProjectList',
  props: ['projectList'],
  components: {
    Project,
  },
};
</script>

<style scoped>
</style>
