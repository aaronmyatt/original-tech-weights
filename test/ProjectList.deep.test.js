import { mount } from 'vue-test-utils'
import ProjectList from '../src/components/ProjectList'
import Project from '../src/components/Project'

describe('ProjectList', () => {
  let cmp

  beforeEach(() => {
    cmp = mount(ProjectList, {
        propsData: {
            projectList: [
                {
                    projectName: 'Test Project',
                    projectWeights: [{
                      name: 'Test',
                      weight: 100,
                    }]
                },
            ],
        },
    })
  })

  it('has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot()
  })

  describe('Project Component Dependency', () => {

      it('renders a list of Project components', () => {
          expect(cmp.findAll('.project-list .project').contains(Project)).toBe(true)
      })

      it('passes props to sub components', () => {
          expect(cmp.find('.project-list .project-name').text().trim()).toBe(cmp.vm.projectList[0].projectName)
      })

  })

})
