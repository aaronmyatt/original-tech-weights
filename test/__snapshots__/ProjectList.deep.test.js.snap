// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectList has the expected html structure 1`] = `
<ul
  class="project-list"
>
  <li
    class="project card"
  >
    <ul
      class="project-container list-group"
    >
      <li
        class="project-name card-title list-group-item d-flex w-100 justify-content-between"
      >
        <h5>
          Test Project
        </h5>
      </li>
       
      <li
        class="project-description list-group-item d-flex w-100 justify-content-between"
      >
        
    
  
      </li>
       
      <li
        class="project-date list-group-item d-flex w-100 justify-content-between"
      >
        
    
  
      </li>
       
      <li
        class="project-owner list-group-item d-flex w-100 justify-content-between"
      >
        
    
  
      </li>
       
      <li
        class="project-repo-url list-group-item d-flex w-100 justify-content-between"
      >
        
    
  
      </li>
       
      <li
        class="project-tech-weights list-group-item"
      >
        <div
          class="weight-list progress"
          style="height: 20px;"
        >
          <div
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="25"
            class="weight progress-bar"
            role="progressbar"
            style="width: 100%; background-color: green;"
          >
            
    Test: 100%
  
          </div>
        </div>
      </li>
    </ul>
  </li>
</ul>
`;
